File: backend/WebApi.Model/Material.cs
namespace WebApi.Model;

public class MaterialInfo
{
    public int MaterialId { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public int PrecioPuntos { get; set; }
    public string? ImagenUrl { get; set; }
}

public class InventarioMaterialInfo
{
    public int MaterialId { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public string? ImagenUrl { get; set; }
    public int Cantidad { get; set; }
}

File: mobile/lib/models/material.dart
class MaterialTienda {
  final int materialId;
  final String nombre;
  final String? descripcion;
  final int precioPuntos;
  final String? imagenUrl;

  MaterialTienda({
    required this.materialId,
    required this.nombre,
    this.descripcion,
    required this.precioPuntos,
    this.imagenUrl,
  });

  factory MaterialTienda.fromJson(Map<String, dynamic> json) {
    return MaterialTienda(
      materialId: json['materialId'],
      nombre: json['nombre'],
      descripcion: json['descripcion'],
      precioPuntos: json['precioPuntos'],
      imagenUrl: json['imagenUrl'],
    );
  }
}

class MaterialInventario {
  final int materialId;
  final String nombre;
  final String? descripcion;
  final String? imagenUrl;
  final int cantidad;

  MaterialInventario({
    required this.materialId,
    required this.nombre,
    this.descripcion,
    this.imagenUrl,
    required this.cantidad,
  });

  factory MaterialInventario.fromJson(Map<String, dynamic> json) {
    return MaterialInventario(
      materialId: json['materialId'],
      nombre: json['nombre'],
      descripcion: json['descripcion'],
      imagenUrl: json['imagenUrl'],
      cantidad: json['cantidad'],
    );
  }
}